osf/quickfiles/models.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.core.exceptions import ValidationError
+
+
+from django.contrib.contenttypes.models import ContentType
+from osf.models import BaseFileNode, Folder, TrashedFileNode, OSFUser
+from osf.models.mixins import CleanMixin
+
+
+class QuickFileNode(BaseFileNode):
+    _provider = 'osfstorage'
+
+
+class QuickFolder(QuickFileNode, Folder, CleanMixin):
+    """
+    QuickFolder's a are specialized OsfStorageFolder models that are attached to the user instead of a
+    project, preprint, registration etc. QuickFolders have a few special restrictions normal Folder models don't have.
+
+    1. They must have a flat file structure. Quickfolders can have no parents nor can it's children be folders.
+    2. They must be attached to a user on creation, meaning the their target must always be a OSFUser instance.
+    3. If a user is deleted, disabled or marked as spam that user's quickfiles must become inaccessible.
+    4. Quickfiles must appear in ES results unless the user is deleted, disabled or marked as spam.
+    5. When loggable events (file uploads, renames .etc) happen to Quickfiles they are logged with that user's info,
```

Comment:
Users were already associated with NodeLogs - as they are a foreignkey on NodeLogs. What's unique about these logs is that they're done `on` the user . Are we going to need to distinguish logs on the user versus logs by the user?  For example, if you check a user's logs in the shell on develop, you see project created, tag added, contributor added, registration created.